cpp/arcticdb/version/version_utils.hpp

```diff
@@ -175,21 +191,25 @@ inline std::optional<RefKey> get_symbol_ref_key(
     return std::make_optional(std::move(ref_key));
 }
 
-inline void read_symbol_ref(std::shared_ptr<StreamSource> store, const StreamId &stream_id, VersionMapEntry &entry) {
+inline void read_symbol_ref(const std::shared_ptr<StreamSource>& store, const StreamId &stream_id, VersionMapEntry &entry) {
     auto maybe_ref_key = get_symbol_ref_key(store, stream_id);
     if (!maybe_ref_key)
         return;
 
     auto [key, seg] = store->read_sync(maybe_ref_key.value());
-    VersionId version_id{};
-    std::tie(entry.head_, version_id) = read_segment_with_keys(seg, entry);
+    LoadProgress load_progress;
+    entry.head_ = read_segment_with_keys(seg, entry, load_progress);
 }
 
 inline void write_symbol_ref(std::shared_ptr<StreamSink> store,
                              const AtomKey &latest_index,
+                             const std::optional<AtomKey>& previous_key,
                              const AtomKey &journal_key) {
     check_is_index_or_tombstone(latest_index);
     check_is_version(journal_key);
+    if(previous_key)
+        check_is_index_or_tombstone(*previous_key);
```

Comment:
If `previous_key` is an index key, rather than a version key, how would we iterate backwards? Can you detministically go from index->version?!